src/main/drivers/gyro_sync.c

```diff
@@ -0,0 +1,91 @@
+/*
+ * gyro_sync.c
+ *
+ *  Created on: 3 aug. 2015
+ *      Author: borisb
+ */
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdlib.h>
+
+#include "platform.h"
+#include "build_config.h"
+
+#include "common/axis.h"
+#include "common/maths.h"
+
+#include "sensor.h"
+#include "accgyro.h"
+
+#include "accgyro_mpu6050.h"
+#include "accgyro_spi_mpu6000.h"
+#include "sensors/sensors.h"
+#include "sensors/acceleration.h"
+#include "gyro_sync.h"
+
+#include "config/runtime_config.h"
+#include "config/config.h"
+
+bool gyroIsUpdated = false;
+
+uint32_t targetLooptime;
+static uint8_t mpuDivider;
+
+void setGyroRefreshRate(uint32_t looptime, uint8_t lpf) {
+#ifdef STM32F303xC
+    if (lpf == INV_FILTER_256HZ_NOLPF2) {
+        if (!sensors(SENSOR_ACC)) {
+            mpuDivider = constrain(looptime / 125, 4, 32) - 1;              // max sampling rate of 2khz when acc to none
+            targetLooptime = 125 * (mpuDivider + 1);
+        }
+        mpuDivider = constrain(looptime / 125, 5, 32) - 1;                  // max sampling rate of 1.6khz, lowest 250hz
+        targetLooptime = 125 * (mpuDivider + 1);
+    }
+
+    else {
+        mpuDivider = constrain(looptime / 1000, 1, 4) - 1;                  // full sampling rate of 1khz
+        targetLooptime = 1000 * (mpuDivider + 1);
+    }
+#elif STM32F10X
+    if ((!sensors(SENSOR_ACC)) && (lpf == INV_FILTER_256HZ_NOLPF2)) {
+        mpuDivider = constrain(looptime / 125, 4, 32) - 1;                  // max sampling rate of 2khz, lowest 250hz
+        targetLooptime = 125 * (mpuDivider + 1);
+    }
+
+    else if ((sensors(SENSOR_ACC)) && (lpf == INV_FILTER_256HZ_NOLPF2)) {   // max sampling rate of 666,6hz
+        mpuDivider = constrain(looptime / 125, 12, 32) - 1;                 // F1 boards should be able to keep up
+        targetLooptime = 125 * (mpuDivider + 1);
+    }
+
+    else if (!(sensors(SENSOR_ACC)) && !(lpf == INV_FILTER_256HZ_NOLPF2)) { // full sampling rate of 1khz
+        mpuDivider = constrain(looptime / 1000, 1, 4) - 1;                  // when LPF enabled and acc set to none
+        targetLooptime = 1000 * (mpuDivider + 1);
+    }
+
+    else {
+        mpuDivider = constrain(looptime / 1000, 2, 4) - 1;    // max sampling rate of 500hz.
+        targetLooptime = 1000 * (mpuDivider + 1);
+    }
+#else    // This is the code for non MPU60XX based boards. Those are using the regular looptime (min looptime 500)
+    if (looptime > GYRO_READ_PENALTY) {
+    	targetLooptime = looptime - GYRO_READ_PENALTY;
+    }
+    else {
+    	targetLooptime = 0;
+    }
+#endif
+}
+
+void checkIfGyroIsUpdated(void) {
+#ifdef USE_GYRO_MPU6050
```

Comment:
IMO both USE_GYRO_MPU6050 and USE_GYRO_MPU6000 may be defined (and autodetection rules used to select correct one).
Using dispatch mechanism through function pointer is consistent way to handle this: https://github.com/cleanflight/cleanflight/blob/master/src/main/drivers/accgyro.h#L24 (and use dummy function or null handling for other drivers)